modules/solid_mechanics/src/kernels/StressDivergenceRZ.C

```diff
@@ -67,6 +123,75 @@ StressDivergenceRZ::computeQpResidual()
   return div;
 }
 
+void
+StressDivergenceRZ::computeJacobian()
+{
+  DenseMatrix<Number> & ke = _assembly.jacobianBlock(_var.number(), _var.number());
+  _local_ke.resize(ke.m(), ke.n());
+  _local_ke.zero();
+
+  if (_volumetric_locking_correction)
+  {
+    Real vol = 0.0;
+    // calculate volume averaged value of shape function derivative
+    _avg_grad_test.resize(_test.size());
+    for (_i = 0; _i < _test.size(); _i++)
+    {
+      _avg_grad_test[_i].resize(2);
+      _avg_grad_test[_i][_component] = 0.0;
+      vol = 0.0;
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        if (_component == 0)
+          _avg_grad_test[_i][_component] += (_grad_test[_i][_qp](_component) + _test[_i][_qp] / _q_point[_qp](0)) * _JxW[_qp] * _coord[_qp];
+        else
+          _avg_grad_test[_i][_component] += _grad_test[_i][_qp](_component) * _JxW[_qp] * _coord[_qp];
+
+        vol += _JxW[_qp] * _coord[_qp];
+      }
+      _avg_grad_test[_i][_component] /= vol;
```

Comment:
I have used _current_elem_volume instead of _current_elem->volume(). 
_current_elem->volume() does not contain the coordinate system correction.